packages/core/src/hooks/useUniqueId.ts

```diff
@@ -1,6 +1,25 @@
-import { useMemo } from "react";
+import * as React from "react"; // this can be optimized when react 17 support is dropped
 
 import uniqueId from "lodash/uniqueId";
 
-export const useUniqueId = (id?: string, idPrefix?: string) =>
-  useMemo(() => id || uniqueId(idPrefix), [id, idPrefix]);
+/** Credit: https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx
+ *  Modified slightly to suit our purposes.
+ */
+// We `toString()` to prevent bundlers from trying to `import { useId } from 'react';`
+const useReactId = (React as any)["useId".toString()] || (() => undefined);
+
+export const useUniqueId = (
+  deterministicId?: string,
+  idPrefix?: string
+): string => {
+  const [id, setId] = React.useState<string | undefined>(useReactId());
+
+  React.useMemo(
+    () => {
+      if (!id) setId(uniqueId(idPrefix));
+    }, // loadash unique id can be removed when react 17 support is dropped
+    [id, idPrefix]
+  );
```

Comment:
- `useMemo` means we'll have a different (hence unstable) id in the server-side, right?
  - this should be an effect
  - we're not saving the memo value anywhere
- we shouldn't need the `id` dependency

- can't we get rid of the the `lodash` dependency here? Radix's counter solution looks nice and simple

Basically, can't this just be the same as Radix's example?

Reply:
I left `loadash` just to avoid any breaking changes but it should be ok. I'm updating it now.